From daa90b12665f709ae42157494ab080739b17d4c4 Mon Sep 17 00:00:00 2001
From: Someone Serge <sergei.kozlukov@aalto.fi>
Date: Tue, 7 Nov 2023 00:04:22 +0000
Subject: [PATCH 4/5] cli_args: add --extra-path

For extending search paths ad hoc, as required for splayed installations
---
 comfy/cli_args.py | 1 +
 main.py           | 3 +++
 2 files changed, 4 insertions(+)

diff --git a/comfy/cli_args.py b/comfy/cli_args.py
index e79b89c..2e2e823 100644
--- a/comfy/cli_args.py
+++ b/comfy/cli_args.py
@@ -98,6 +98,7 @@ parser.add_argument("--quick-test-for-ci", action="store_true", help="Quick test
 parser.add_argument("--windows-standalone-build", action="store_true", help="Windows standalone build: Enable convenient things that most people using the standalone windows build will probably enjoy (like auto opening the page on startup).")
 
 parser.add_argument("--disable-metadata", action="store_true", help="Disable saving prompt metadata in files.")
+parser.add_argument("--extra-path", nargs=2, action="append", help="Extra search paths for configs and models, e.g. --extra-path checkpoints models/checkpoints/")
 
 if comfy.options.args_parsing:
     args = parser.parse_args()
diff --git a/main.py b/main.py
index 426359e..6e5c303 100644
--- a/main.py
+++ b/main.py
@@ -148,6 +148,9 @@ def main():
         folder_paths.set_temp_directory(temp_dir)
     cleanup_temp()
 
+    for kind, path in args.extra_paths or []:
+        folder_paths.add_model_folder_path(kind, path)
+
     loop = asyncio.new_event_loop()
     asyncio.set_event_loop(loop)
     server = server.PromptServer(loop)
-- 
2.42.0

